static/gsApp/components/profiling/alerts.tsx

```diff
@@ -378,6 +378,169 @@ export function ContinuousProfilingBetaSDKAlertBanner() {
   );
 }
 
+interface ContinuousProfilingNoQuotaAlertBannerProps {
+  organization: Organization;
+  subscription: Subscription;
+}
+
+function ContinuousProfilingNoQuotaAlertAm2Banner({
+  organization,
+  subscription,
+}: ContinuousProfilingNoQuotaAlertBannerProps) {
+  const hasBudget = hasBudgetForContinuousOrUiProfilesHours(subscription);
+
+  // As long as they have a budget set, don't show this banner.
+  if (hasBudget) {
+    return null;
+  }
+
+  // For am2 plans, it's possible they're just using transaction profiles which uses performance units
+  // in this case, we don't want to ask them to set up on-demand budgets for no reason.
+  // So check to see if they're trying to send continuous/ui profiles before shwoing them this banner.
+  if (
+    !subscription.categories.profileDuration?.usageExceeded &&
+    !subscription.categories.profileDurationUI?.usageExceeded
```

Comment:
i believe customers without PAYG that aren't using transaction profiles will fall under this condition too since having usage is not exceeded for profile hour categories unless they have PAYG set up _and_ it's all used up

Reply:
I'm not sure i understand what you're saying here.

Reply:
After some thought, I think this is fine. If the user hasn't sent any kind of profiles, showing them an upsell here is too eager as they can still use transaction profiles.